### PR TITLE
Added CVE-2021-3584

### PR DESCRIPTION
**Pull request by lzap**
SSIA

### CODE REVIEW

File: security.md [hunk 1]
```diff
@@ -85,6 +86,16 @@ All security advisories made for Foreman are listed below with their correspondi
...
 ### Disclosure details
 
+#### <a id="2021-3584"></a>CVE-2021-3584: Remote code execution through Sendmail configuration
+
+Sendmail location and arguments, available via Administer - Settings, both accept arbitrary strings and pass them into shell. By default, only Foreman super administrator can access settings.
+
+*Mitigation:* Verify the both settings and remove edit_settings permissions to all roles and users until fixed. Alternatively, create settings named sendmail_location and sendmail_arguments in `settings.yaml` file to override the UI and make the values read-only.
+
+* Affects Foreman veersion 1.15 and higher
+* Fix released in Foreman 2.4.1, 2.5.1 and higher
```
Comment (lzap) [inline security.md:99]: To be confirmed by @upadhyeammit

File: security.md [hunk 2]
```diff
@@ -85,6 +86,16 @@ All security advisories made for Foreman are listed below with their correspondi
 
 ### Disclosure details
 
+#### <a id="2021-3584"></a>CVE-2021-3584: Remote code execution through Sendmail configuration
+
+Sendmail location and arguments, available via Administer - Settings, both accept arbitrary strings and pass them into shell. By default, only Foreman super administrator can access settings.
```
Comment (ekohl) [inline security.md]:
```suggestion
Sendmail location and arguments, available via Administer - Settings, both accept arbitrary strings and pass them into shell. By default, only Foreman super administrators can access settings.
```

File: security.md [hunk 3]
```diff
@@ -85,6 +86,16 @@ All security advisories made for Foreman are listed below with their correspondi
...
+
+*Mitigation:* Verify the both settings and remove edit_settings permissions to all roles and users until fixed. Alternatively, create settings named sendmail_location and sendmail_arguments in `settings.yaml` file to override the UI and make the values read-only.
```
Comment (ekohl) [inline security.md]:
```suggestion
*Mitigation:* Verify the both settings and remove edit_settings permissions from all roles and users until fixed. Alternatively, create settings named `sendmail_location` and `sendmail_arguments` in `settings.yaml` file to override the UI and make the values read-only. Note that the installer will remove these again.
```

File: security.md [hunk 4]
```diff
@@ -85,6 +86,16 @@ All security advisories made for Foreman are listed below with their correspondi
...
+
+* Affects Foreman veersion 1.15 and higher
```
Comment (ekohl) [inline security.md]:
```suggestion
* Affects Foreman version 1.15 and higher
```

**Review by ekohl — CHANGES_REQUESTED** [review]

File: security.md (hunk 1 above)
Comment (lzap) [reply security.md:99]: Confirmed by Amit thanks.

File: security.md (hunk 4 above)
Comment (lzap) [reply security.md]: Looks quite Dutch :-) veersion 🇳🇱

**Comment by lzap** [comment]
Oh batch commit feature is nice, please squash on merge.

File: security.md (hunk 4 above)
Comment (ekohl) [reply security.md]: The Dutch translation is `versie`. `veer` means feather or spring (the metal kind that you can compress and it'll take its old shape again) but there's no `veersion` ;)

File: security.md [hunk 5]
```diff
@@ -85,6 +86,16 @@ All security advisories made for Foreman are listed below with their correspondi
 
 ### Disclosure details
 
+#### <a id="2021-3584"></a>CVE-2021-3584: Remote code execution through Sendmail configuration
+
+Sendmail location and arguments, available via Administer - Settings, both accept arbitrary strings and pass them into shell. By default, only Foreman super administrators can access settings.
+
+*Mitigation:* Verify the both settings and remove edit_settings permissions from all roles and users until fixed. Alternatively, create settings named `sendmail_location` and `sendmail_arguments` in `settings.yaml` file to override the UI and make the values read-only. Note that the installer will remove these again.
```
Comment (ekohl) [inline security.md]: Just wondering: will users understand that they really should use `:sendmail_location`? Perhaps an example section is good?

Comment (tbrisker) [inline security.md]:
```suggestion
*Mitigation:* Verify the value of both settings and remove the `edit_settings` permissions from any untrusted users until fixed. Alternatively, create settings named `sendmail_location` and `sendmail_arguments` in `settings.yaml` file to override the UI and make the values read-only. Note that the installer will remove these again.
```

File: security.md [hunk 6]
```diff
@@ -85,6 +86,16 @@ All security advisories made for Foreman are listed below with their correspondi
 
 ### Disclosure details
 
+#### <a id="2021-3584"></a>CVE-2021-3584: Remote code execution through Sendmail configuration
+
+Sendmail location and arguments, available via Administer - Settings, both accept arbitrary strings and pass them into shell. By default, only Foreman super administrators can access settings.
```
Comment (tbrisker) [inline security.md]:
```suggestion
Sendmail location and arguments, available via Administer - Settings, both accept arbitrary strings and pass them into shell. By default, Foreman administrators as well as "Manager" and "System Administrator" roles can modify settings.
```

**Comment by lzap** [comment]
Added example, please squash.

**Comment by ohadlevy** [comment]
@lzap no need to ack the reporter?

**Comment by lzap** [comment]
Good catch, amended. Cheers @ohadlevy ! :-)

File: security.md (hunk 4 above)
Comment (ekohl) [reply security.md]: Typo is still here

Comment (lzap) [reply security.md]: Damn, rebased.

File: security.md (hunk 5 above)
Comment (ekohl) [reply security.md]: I still think this is better. Otherwise users may create an entry that ends up being unused. A literal example is even better.

Comment (lzap) [reply security.md]: Done.

File: security.md (hunk 2 above)
Comment (ekohl) [reply security.md]: Apologies for not catching this the first time
```suggestion
Sendmail location and arguments, available via Administer - Settings, both accept arbitrary strings and pass them into shell. By default, only Foreman super administrators can access settings. However, non-admin users can be granted the permission to edit settings.
```

File: security.md [hunk 7]
```diff
@@ -85,6 +86,21 @@ All security advisories made for Foreman are listed below with their correspondi
...
 ### Disclosure details
 
+#### <a id="2021-3584"></a>CVE-2021-3584: Remote code execution through Sendmail configuration
+
+Sendmail location and arguments, available via Administer - Settings, both accept arbitrary strings and pass them into shell. By default, only Foreman super administrator can access settings.
+
+*Mitigation:* Verify the both settings and remove edit_settings permissions to all roles and users until fixed. Alternatively, create entries in `settings.yaml` to override UI settings and make the read-only:
+
+    :sendmail_location: "/usr/sbin/sendmail"
+    :sendmail_arguments: "-i"
```
Comment (ekohl) [inline security.md:96]:
````suggestion
*Mitigation 1:* Verify the both settings and remove edit_settings permissions to all roles and users until fixed.

*Mitigation 2:* Create entries in `settings.yaml` to override UI settings and make them read-only. Note that the installer will revert these changes. [#32827](https://projects.theforeman.org/issues/32827) will add options to the installer which makes it supported.

```yaml
:sendmail_location: "/usr/sbin/sendmail"
:sendmail_arguments: "-i"
```
````

**Comment by lzap** [comment]
Merged thanks!

File: security.md (hunk 2 above)
Comment (tbrisker) [reply security.md]: This is incorrect. By default users with the `Manager` or `System admin` roles also have permission to edit settings.

Comment (lzap) [reply security.md]: Corrected.

**Comment by lzap** [comment]
Rebased.

**Comment by lzap** [comment]
@ekohl mind finishing this up? Thanks!

**Pull request closed without merging**